source/common/http/mixed_conn_pool.h

```diff
@@ -0,0 +1,33 @@
+#pragma once
+
+#include "common/http/conn_pool_base.h"
+
+namespace Envoy {
+namespace Http {
+
+// An HTTP connection pool which supports both HTTP/1 and HTTP/2 based on ALPN
+class HttpConnPoolImplMixed : public HttpConnPoolImplBase {
+public:
+  HttpConnPoolImplMixed(Event::Dispatcher& dispatcher, Random::RandomGenerator& random_generator,
+                        Upstream::HostConstSharedPtr host, Upstream::ResourcePriority priority,
+                        const Network::ConnectionSocket::OptionsSharedPtr& options,
+                        const Network::TransportSocketOptionsSharedPtr& transport_socket_options,
+                        Upstream::ClusterConnectivityState& state)
+      : HttpConnPoolImplBase(std::move(host), std::move(priority), dispatcher, options,
+                             transport_socket_options, random_generator, state,
+                             {Protocol::Http2, Protocol::Http11}) {}
+
+  Envoy::ConnectionPool::ActiveClientPtr instantiateActiveClient() override;
+  CodecClientPtr createCodecClient(Upstream::Host::CreateConnectionData& data) override;
+
+  void onConnected(Envoy::ConnectionPool::ActiveClient& client) override;
+  Http::Protocol protocol() { return protocol_; }
+
+private:
+  bool connected_{};
```

Comment:
What does connected_ mean at the conn pool level? Better name and/or comment?

Reply:
Is this used? Do you maybe mean to guard re-checking protocol if we previously had a connection? If so should this be `saw_first_connection_` or something like that?